rules/falco_rules.yaml

```diff
@@ -3420,3 +3420,38 @@
 # Application rules have moved to application_rules.yaml. Please look
 # there if you want to enable them by adding to
 # falco_rules.local.yaml.
+
+
+- macro: rocketmq_check_netty_server
+  condition: (proc.name contains NettyServer and proc.env contains ROCKETMQ)
+
+- macro: rocketmq_check_payload_for_conf_update
+  condition: (evt.type=read and evt.buffer contains 'code":25' and evt.buffer contains serializeTypeCurrentRPC and evt.buffer contains JSON)
```

Comment:
Have you had the opportunity to test this detection in a production environment? If yes have you noticed changes in resource utilization and event drops?

Context: I/O syscalls especially `read` and `write` are quite heavy and the primary concern is that for most adopters the event drop could drastically increase. For this reason existing read and write macros have been disabled https://github.com/falcosecurity/rules/blob/main/rules/falco_rules.yaml#L22C2-L28

All this being said we need to find better ways to support these types of detections, it's a current tooling limitation we are painfully aware.

We could consider disabling this detection by default. CC @falcosecurity/rules-maintainers

Reply:
> Have you had the opportunity to test this detection in a production environment? If yes have you noticed changes in resource utilization and event drops?q

I have not tested it in a production env. I was looking for a helping hand from the community to make it tested & validated in real prod or beta environments. 

> Context: I/O syscalls especially `read` and `write` are quite heavy and the primary concern is that for most adopters the event drop could drastically increase. For this reason existing read and write macros have been disabled https://github.com/falcosecurity/rules/blob/main/rules/falco_rules.yaml#L22C2-L28

I agree. Unfortunately, the payload was captured only in `read` syscall. Initially, I tried to develop a detection checking the `evt.buffer` without using '-A' switch but however we could not determine the pattern of successful exploitation. 

Would you suggest any workaround where else we can capture the payload? The exploit happens through socket but we did not see much in any sock-based syscalls. 

Another thought is to capture relevant syscalls and compare patterns in the fixed version vs the vulnerable version. This approach would kill a lot of time and we cannot scale it for a large number of detections. 

> All this being said we need to find better ways to support these types of detections, it's a current tooling limitation we are painfully aware.
> 
> We could consider disabling this detection by default. CC @falcosecurity/rules-maintainers

It makes sense to disable it by default and also using a disclaimer in a comment would caution users.

Reply:
> Would you suggest any workaround where else we can capture the payload?

Let's see when I have time for a hacking session to explore, but it seems you already explored various avenues thoroughly. Likely it is something we need to put on the longer-term roadmap to see what else could be done. For Falco 0.36 symlink resolution is prioritized for the next kernel driver revamp.

Would you want to open an issue either here or in the Falco repo for tracking? Would appreciate it!

---


Mind squashing the initial commit and re-pushing with a signoff 🙃 thank you!